components/Events/Filters.js

```diff
@@ -0,0 +1,204 @@
+
+export default function filters(props) {
+  const [isFilterOpen, setIsFilterOpen] = useState(false);
+  const [searchValue, setSearchValue] = useState('');
+  const [chosenComms, setChosenComms] = useState({
+    studio: false,
+    icpc: false,
+    design: false,
+    cyber: false,
+    teachLA: false,
+    w: false,
+    ai: false,
+    hack: false,
+    board: false,
+  });
+  const [chosenDays, setChosenDays] = useState({
+    Sunday: false,
+    Monday: false,
+    Tuesday: false,
+    Wednesday: false,
+    Thursday: false,
+    Friday: false,
+    Saturday: false,
+  });
+  const [fromTime, setFromTime] = useState('');
+  const [toTime, setToTime] = useState('');
+  const [chosenLoc, setChosenLoc] = useState({
+    Online: false,
+    In_person: false,
+  });
+
+  const handleSearch = () => {
+    let allEvents = [...events];
+    //Check search input
+    const lowerCaseSearch = searchValue.toLowerCase();
+    allEvents = allEvents.filter((item) => {
+      if (String(item.title).toLowerCase().includes(lowerCaseSearch))
+        return true;
+      if (String(item.committee).toLowerCase().includes(lowerCaseSearch))
+        return true;
+      if (String(item.location).toLowerCase().includes(lowerCaseSearch))
+        return true;
+      if (String(item.description).toLowerCase().includes(lowerCaseSearch))
+        return true;
+      return false;
+    });
+    //Check filters
+    const isTimeEmpty = fromTime === '' || toTime === '';
+    const isCommEmpty = Object.keys(chosenComms).every(
+      (comm) => !chosenComms[comm],
+    );
+    const isDaysEmpty = Object.keys(chosenDays).every(
+      (day) => !chosenDays[day],
+    );
+    const isLocEmpty = Object.keys(chosenLoc).every((loc) => !chosenLoc[loc]);
+    let fromMins = 0;
+    let toMins = 0;
+    if (!isTimeEmpty) {
+      fromMins =
+        parseInt(fromTime.split(':')[0]) * 60 +
+        parseInt(fromTime.split(':')[1]);
+      toMins =
+        parseInt(toTime.split(':')[0]) * 60 + parseInt(toTime.split(':')[1]);
+    }
+    allEvents = allEvents.filter((item) => {
+      const itemDate = new Date(item.start);
+      const itemDay = Object.keys(chosenDays)[itemDate.getDay()];
+      const itemLoc = item.location.toLowerCase().includes('zoom') ? 'Online' : 'In_person';
+      if (!item.committee) return false;
+      const itemComm =
+        item.committee === 'teach-la' ? 'teachLA' : item.committee; //No hyphens in object key
+      const itemMins = itemDate.getHours() * 60 + itemDate.getMinutes();
+      const timeValid =
+        isTimeEmpty || (itemMins >= fromMins && itemMins <= toMins);
+      const dayValid = isDaysEmpty || chosenDays[itemDay];
+      const commValid = isCommEmpty || chosenComms[itemComm];
+      const onlineValid = isLocEmpty || chosenLoc[itemLoc];
+      return timeValid && dayValid && commValid && onlineValid;
+    });
+    const newEvents = allEvents.map((original_event, index) => ({
+      ...original_event,
+      id: index,
+    }));
```

Comment:
Could we separate out each filter so that it's easy for people to add filters in the future if they want + for code readability? EG (I didn't test so it might not be right)
```
    // Check time filter
    const isFilterTimeEmpty = Object.keys(chosenTimes).every(
      (time) => time === ""
    );
    if (!isFilterTimeEmpty) {
      filterFromMins =
        parseInt(fromTime.split(":")[0]) * 60 +
        parseInt(fromTime.split(":")[1]);
      filterToMins =
        parseInt(toTime.split(":")[0]) * 60 + parseInt(toTime.split(":")[1]);

      allEvents = allEvents.filter((item) => {
        const itemStartDate = new Date(item.start);
        const itemStartMins =
          itemStartDate.getHours() * 60 + itemStartDate.getMinutes();

        const itemEndDate = new Date(item.end);
        const itemEndMins =
          itemEndDate.getHours() * 60 + itemEndDate.getMinutes();

        const timeValid =
          isTimeEmpty ||
          (itemStartMins >= filterFromMins && itemEndMins <= filterToMins);
        return timeValid;
      });
    }
```

Reply:
I also wouldn't mind putting
```
        const itemEndDate = new Date(item.end);
        const itemEndMins =
          itemEndDate.getHours() * 60 + itemEndDate.getMinutes();
```
in a separate method like `dateToMinutes` or something like that